be/src/exec/pipeline/runtime_filter_types.h

```diff
@@ -0,0 +1,336 @@
+
+#pragma once
+#include <memory>
+#include <mutex>
+
+#include "common/statusor.h"
+#include "exprs/expr_context.h"
+#include "exprs/predicate.h"
+#include "exprs/vectorized/runtime_filter_bank.h"
+
+namespace starrocks {
+namespace pipeline {
+class RuntimeFilterHolder;
+using RuntimeFilterHolderPtr = std::unique_ptr<RuntimeFilterHolder>;
+using RuntimeInFilter = starrocks::ExprContext;
+using RuntimeBloomFilter = starrocks::vectorized::RuntimeFilterBuildDescriptor;
+using RuntimeBloomFilterProbeDescriptor = starrocks::vectorized::RuntimeFilterProbeDescriptor;
+using RuntimeBloomFilterProbeDescriptorPtr = RuntimeBloomFilterProbeDescriptor*;
+using RuntimeBloomFilterRunningContext = starrocks::vectorized::JoinRuntimeFilter::RunningContext;
+using RuntimeInFilterPtr = RuntimeInFilter*;
+using RuntimeBloomFilterPtr = RuntimeBloomFilter*;
+using RuntimeInFilters = std::list<RuntimeInFilterPtr>;
+using RuntimeBloomFilters = std::list<RuntimeBloomFilterPtr>;
+struct RuntimeFilterCollector;
+using RuntimeFilterCollectorPtr = std::unique_ptr<RuntimeFilterCollector>;
+using RuntimeFilterProbeCollector = starrocks::vectorized::RuntimeFilterProbeCollector;
+using Predicate = starrocks::Predicate;
+struct RuntimeBloomFilterBuildParam;
+using RuntimeBloomFilterBuildParams = std::list<RuntimeBloomFilterBuildParam>;
+// Parameters used to build runtime bloom-filters.
+struct RuntimeBloomFilterBuildParam {
+    RuntimeBloomFilterBuildParam(bool eq_null, const ColumnPtr& column, size_t column_offset, size_t ht_row_count)
+            : eq_null(eq_null), column(column), column_offset(column_offset), ht_row_count(ht_row_count) {}
+    bool eq_null;
+    ColumnPtr column;
+    size_t column_offset;
+    size_t ht_row_count;
+};
+
+// RuntimeFilterCollector contains runtime in-filters and bloom-filters, it is stored in RuntimeFilerHub
+// and every HashJoinBuildOperatorFactory has its corresponding RuntimeFilterCollector.
+struct RuntimeFilterCollector {
+    RuntimeFilterCollector(RuntimeInFilters&& in_filters, RuntimeBloomFilters&& bloom_filters)
+            : _in_filters(std::move(in_filters)), _bloom_filters(std::move(bloom_filters)) {}
+
+    RuntimeBloomFilters& get_bloom_filters() { return _bloom_filters; }
+    RuntimeInFilters& get_in_filters() { return _in_filters; }
+
+    std::vector<RuntimeInFilterPtr> get_in_filters_bounded_by_tuple_ids(const std::vector<TupleId>& tuple_ids) {
+        std::vector<ExprContext*> selected_in_filters;
+        for (auto* in_filter : _in_filters) {
+            if (in_filter->root()->is_bound(tuple_ids)) {
+                selected_in_filters.push_back(in_filter);
+            }
+        }
+        return selected_in_filters;
+    }
+
+private:
+    // local runtime in-filter
+    RuntimeInFilters _in_filters;
+    // global/local runtime bloom-filter(including max-min filter)
+    RuntimeBloomFilters _bloom_filters;
+};
+
+class RuntimeFilterHolder {
```

Comment:
Why need `RuntimeFilterHolder`? not directly use `RuntimeFilterCollector` ?

Reply:
a placeholder is required,  that placeholder contains nullptr means that the runtime filter is not ready, otherwise, the runtime filter is ready.